[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xf33512]

go-waku logs:

```
2022-12-23T16:33:06.625+1100	[34mINFO[0m	gowaku	waku/node.go:94	Version details 	{"version": "0.3.0", "commit": "183d782"}
2022-12-23T16:33:06.667+1100	[35mDEBUG[0m	rcmgr	resource-manager/limit.go:84	initializing new limiter with config	{"limits": {"System":{"Streams":10008,"StreamsInbound":5004,"StreamsOutbound":10008,"Conns":625,"ConnsInbound":312,"ConnsOutbound":625,"FD":262144,"Memory":4307550208},"Transient":{"Streams":1251,"StreamsInbound":625,"StreamsOutbound":1251,"Conns":188,"ConnsInbound":94,"ConnsOutbound":188,"FD":65536,"Memory":555220992},"AllowlistedSystem":{"Streams":10008,"StreamsInbound":5004,"StreamsOutbound":10008,"Conns":625,"ConnsInbound":312,"ConnsOutbound":625,"FD":262144,"Memory":4307550208},"AllowlistedTransient":{"Streams":1251,"StreamsInbound":625,"StreamsOutbound":1251,"Conns":188,"ConnsInbound":94,"ConnsOutbound":188,"FD":65536,"Memory":555220992},"ServiceDefault":{"Streams":12056,"StreamsInbound":3014,"StreamsOutbound":12056,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":588775424},"Service":{"libp2p.autonat":{"Streams":79,"StreamsInbound":79,"StreamsOutbound":79,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":12345344},"libp2p.holepunch":{"Streams":126,"StreamsInbound":63,"StreamsOutbound":63,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":20496384},"libp2p.identify":{"Streams":625,"StreamsInbound":312,"StreamsOutbound":312,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":20496384},"libp2p.ping":{"Streams":312,"StreamsInbound":312,"StreamsOutbound":312,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":20496384},"libp2p.relay/v1":{"Streams":1251,"StreamsInbound":1251,"StreamsOutbound":1251,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":81985536},"libp2p.relay/v2":{"Streams":1251,"StreamsInbound":1251,"StreamsOutbound":1251,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":81985536}},"ServicePeerDefault":{"Streams":287,"StreamsInbound":143,"StreamsOutbound":287,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":33079296},"ServicePeer":{"libp2p.autonat":{"Streams":2,"StreamsInbound":2,"StreamsOutbound":2,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":1048576},"libp2p.holepunch":{"Streams":2,"StreamsInbound":2,"StreamsOutbound":2,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":1048576},"libp2p.identify":{"Streams":32,"StreamsInbound":16,"StreamsOutbound":16,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":1048576},"libp2p.ping":{"Streams":4,"StreamsInbound":2,"StreamsOutbound":3,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":8590458880},"libp2p.relay/v1":{"Streams":64,"StreamsInbound":64,"StreamsOutbound":64,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":1048576},"libp2p.relay/v2":{"Streams":64,"StreamsInbound":64,"StreamsOutbound":64,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":1048576}},"ProtocolDefault":{"Streams":4038,"StreamsInbound":1507,"StreamsOutbound":4038,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":735494144},"Protocol":{"/ipfs/id/1.0.0":{"Streams":625,"StreamsInbound":312,"StreamsOutbound":312,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":20496384},"/ipfs/id/push/1.0.0":{"Streams":625,"StreamsInbound":312,"StreamsOutbound":312,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":20496384},"/ipfs/ping/1.0.0":{"Streams":312,"StreamsInbound":312,"StreamsOutbound":312,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":20496384},"/libp2p/autonat/1.0.0":{"Streams":79,"StreamsInbound":79,"StreamsOutbound":79,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":12345344},"/libp2p/circuit/relay/0.1.0":{"Streams":3127,"StreamsInbound":3127,"StreamsOutbound":3127,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":81985536},"/libp2p/circuit/relay/0.2.0/hop":{"Streams":3127,"StreamsInbound":3127,"StreamsOutbound":3127,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":81985536},"/libp2p/circuit/relay/0.2.0/stop":{"Streams":3127,"StreamsInbound":3127,"StreamsOutbound":3127,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":81985536},"/libp2p/dcutr":{"Streams":126,"StreamsInbound":63,"StreamsOutbound":63,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":20496384},"/p2p/id/delta/1.0.0":{"Streams":625,"StreamsInbound":312,"StreamsOutbound":312,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":20496384}},"ProtocolPeerDefault":{"Streams":318,"StreamsInbound":79,"StreamsOutbound":159,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":16777231},"ProtocolPeer":{"/ipfs/id/1.0.0":{"Streams":32,"StreamsInbound":16,"StreamsOutbound":16,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":8590458880},"/ipfs/id/push/1.0.0":{"Streams":32,"StreamsInbound":16,"StreamsOutbound":16,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":8590458880},"/ipfs/ping/1.0.0":{"Streams":4,"StreamsInbound":2,"StreamsOutbound":3,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":8590458880},"/libp2p/autonat/1.0.0":{"Streams":2,"StreamsInbound":2,"StreamsOutbound":2,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":1048576},"/libp2p/circuit/relay/0.1.0":{"Streams":128,"StreamsInbound":128,"StreamsOutbound":128,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":33554432},"/libp2p/circuit/relay/0.2.0/hop":{"Streams":128,"StreamsInbound":128,"StreamsOutbound":128,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":33554432},"/libp2p/circuit/relay/0.2.0/stop":{"Streams":128,"StreamsInbound":128,"StreamsOutbound":128,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":33554432},"/libp2p/dcutr":{"Streams":2,"StreamsInbound":2,"StreamsOutbound":2,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":1048576},"/p2p/id/delta/1.0.0":{"Streams":32,"StreamsInbound":16,"StreamsOutbound":16,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":8590458880}},"PeerDefault":{"Streams":1507,"StreamsInbound":753,"StreamsOutbound":1507,"Conns":8,"ConnsInbound":4,"ConnsOutbound":8,"FD":4096,"Memory":588775424},"Conn":{"Streams":0,"StreamsInbound":0,"StreamsOutbound":0,"Conns":1,"ConnsInbound":1,"ConnsOutbound":1,"FD":1,"Memory":33554432},"Stream":{"Streams":1,"StreamsInbound":1,"StreamsOutbound":1,"Conns":0,"ConnsInbound":0,"ConnsOutbound":0,"FD":0,"Memory":16777216}}}
2022-12-23T16:33:06.682+1100	[34mINFO[0m	gowaku.node2.timesource	timesource/ntp.go:169	starting service	{"node": "16Uiu2HAmCm8NywjfQbG1ncyQUoevLJk5kifG1nj8UJ8JdFAAbniP"}
2022-12-23T16:33:06.682+1100	[34mINFO[0m	gowaku.node2.store	store/waku_store_protocol.go:108	Store protocol started (no message provider)	{"node": "16Uiu2HAmCm8NywjfQbG1ncyQUoevLJk5kifG1nj8UJ8JdFAAbniP"}
2022-12-23T16:33:06.682+1100	[34mINFO[0m	gowaku.node2	node/keepalive.go:23	setting up ping protocol	{"node": "16Uiu2HAmCm8NywjfQbG1ncyQUoevLJk5kifG1nj8UJ8JdFAAbniP", "duration": 300}
2022-12-23T16:33:06.685+1100	[34mINFO[0m	gowaku.node2	node/wakunode2.go:241	listening	{"node": "16Uiu2HAmCm8NywjfQbG1ncyQUoevLJk5kifG1nj8UJ8JdFAAbniP", "multiaddr": ["/ip4/127.0.0.1/tcp/8001/p2p/16Uiu2HAmCm8NywjfQbG1ncyQUoevLJk5kifG1nj8UJ8JdFAAbniP", "/ip4/192.168.178.251/tcp/8002/ws/p2p/16Uiu2HAmCm8NywjfQbG1ncyQUoevLJk5kifG1nj8UJ8JdFAAbniP", "/ip4/127.0.0.1/tcp/8002/ws/p2p/16Uiu2HAmCm8NywjfQbG1ncyQUoevLJk5kifG1nj8UJ8JdFAAbniP"]}
2022-12-23T16:33:06.689+1100	[34mINFO[0m	gowaku.node2	node/localnode.go:258	enr record	{"node": "16Uiu2HAmCm8NywjfQbG1ncyQUoevLJk5kifG1nj8UJ8JdFAAbniP", "enr": "enr:-KO4QAV51W3CaYNtmQPFCmZjA2R65_MT_Knke_kqDrsCVZhGenlIaJmW9HGJNx0QkYci0q1winEB1TXiYvHV-RMQg8EBgmlkgnY0gmlwhMCosvuKbXVsdGlhZGRyc4wACgTAqLL7Bh9C3QOJc2VjcDI1NmsxoQMBkpBAaWd1v7mA7rDhprg-lNyuZh-xDrCAELjsrJ2utIN0Y3CCH0KFd2FrdTIC"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xf33512]

goroutine 63 [running]:
github.com/waku-org/go-waku/waku/v2/protocol/relay.(*WakuRelay).Topics(0x67a6c0?)
	/home/fryorcraken/src/waku-org/go-waku/waku/v2/protocol/relay/waku_relay.go:107 +0x52
github.com/waku-org/go-waku/waku/v2/rpc.(*RelayService).Start(0xc00094fd80)
	/home/fryorcraken/src/waku-org/go-waku/waku/v2/rpc/relay.go:71 +0x3d
created by github.com/waku-org/go-waku/waku/v2/rpc.(*WakuRpc).Start
	/home/fryorcraken/src/waku-org/go-waku/waku/v2/rpc/waku_rpc.go:113 +0x76

```


args:
```
waku:nwaku nwaku args: --listen-address=127.0.0.1 --rpc=true --relay=false --rpc-admin=true --websocket-support=true --log-level=DEBUG --tcp-port=8001 --rpc-port=8000 --websocket-port=8002 --min-relay-peers-to-publish=0
```

DB files (May be the issue):

[db-files.zip](https://github.com/waku-org/go-waku/files/10292335/db-files.zip)

--- a/waku/v2/rest/waku_rest.go
+++ b/waku/v2/rest/waku_rest.go
@@ -33,7 +33,6 @@ func NewWakuRest(node *node.WakuNode, address string, port int, enableAdmin bool
 	}
 
 	_ = NewDebugService(node, mux)
-	relayService := NewRelayService(node, mux, relayCacheCapacity, log)
 
 	listenAddr := fmt.Sprintf("%s:%d", address, port)
 
@@ -42,20 +41,27 @@ func NewWakuRest(node *node.WakuNode, address string, port int, enableAdmin bool
 		Handler: mux,
 	}
 
-	server.RegisterOnShutdown(func() {
-		relayService.Stop()
-	})
-
 	wrpc.node = node
 	wrpc.server = server
-	wrpc.relayService = relayService
+
+	if node.Relay() != nil {
+		relayService := NewRelayService(node, mux, relayCacheCapacity, log)
+		server.RegisterOnShutdown(func() {
+			relayService.Stop()
+		})
+		wrpc.relayService = relayService
+	}
 
 	return wrpc
 }
 
 func (r *WakuRest) Start(ctx context.Context, wg *sync.WaitGroup) {
 	defer wg.Done()
-	go r.relayService.Start(ctx)
+
+	if r.node.Relay() != nil {
+		go r.relayService.Start(ctx)
+	}
+
 	go func() {
 		_ = r.server.ListenAndServe()
 	}()

--- a/waku/v2/rpc/waku_rpc.go
+++ b/waku/v2/rpc/waku_rpc.go
@@ -51,10 +51,13 @@ func NewWakuRpc(node *node.WakuNode, address string, port int, enableAdmin bool,
 		wrpc.log.Error("registering debug service", zap.Error(err))
 	}
 
-	relayService := NewRelayService(node, cacheCapacity, log)
-	err = s.RegisterService(relayService, "Relay")
-	if err != nil {
-		wrpc.log.Error("registering relay service", zap.Error(err))
+	var relayService *RelayService
+	if node.Relay() != nil {
+		relayService = NewRelayService(node, cacheCapacity, log)
+		err = s.RegisterService(relayService, "Relay")
+		if err != nil {
+			wrpc.log.Error("registering relay service", zap.Error(err))
+		}
 	}
 
 	err = s.RegisterService(&StoreService{node, log}, "Store")
@@ -95,7 +98,10 @@ func NewWakuRpc(node *node.WakuNode, address string, port int, enableAdmin bool,
 
 	server.RegisterOnShutdown(func() {
 		filterService.Stop()
-		relayService.Stop()
+
+		if relayService != nil {
+			relayService.Stop()
+		}
 		if wrpc.privateService != nil {
 			wrpc.privateService.Stop()
 		}
@@ -110,7 +116,10 @@ func NewWakuRpc(node *node.WakuNode, address string, port int, enableAdmin bool,
 }
 
 func (r *WakuRpc) Start() {
-	go r.relayService.Start()
+	if r.relayService != nil {
+		go r.relayService.Start()
+	}
+
 	go r.filterService.Start()
 	if r.privateService != nil {
 		go r.privateService.Start()

